optuna/samplers/_qmc.py

```diff
@@ -0,0 +1,345 @@
+from collections import OrderedDict
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Sequence
+
+import numpy as np
+import scipy
+
+import optuna
+from optuna import distributions
+from optuna import logging
+from optuna._experimental import experimental
+from optuna._transform import _SearchSpaceTransform
+from optuna.distributions import BaseDistribution
+from optuna.samplers import BaseSampler
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+_logger = logging.get_logger(__name__)
+
+_SUGGESTED_STATES = (TrialState.COMPLETE, TrialState.PRUNED)
+
+_NUMERICAL_DISTRIBUTIONS = (
+    distributions.UniformDistribution,
+    distributions.LogUniformDistribution,
+    distributions.DiscreteUniformDistribution,
+    distributions.IntUniformDistribution,
+    distributions.IntLogUniformDistribution,
+)
+
+
+@experimental("3.0.0a1")
```

Comment:
We decided to specify the official version instead of the alpha version as we discussed [here](https://github.com/optuna/optuna/pull/3054#issuecomment-954433091), because most users will notice this feature in the official version, not the alpha version.
```suggestion
@experimental("3.0.0")
```